### PR TITLE
Add a new behavior for Cmd/Ctrl + Enter keyboard shortcut

### LINKED ISSUE #9366: Add support for Cmd/Ctrl + Enter keyboard shortcut
### Description

The issue is about adding a new keyboard shortcut Cmd/Ctrl + Enter. The issue is part of the [Epic issue](https://github.com/handsontable/handsontable/issues/9361) and here are the expected results:

Shortcut | Expected behavior
-- | --
Cmd/Ctrl + Enter | Fill the selected range of cells with the value of the active cell

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR adds the new Cmd/Ctrl + Enter keyboard shortcuts to Handsontable. Now it works as follows:

Shortcut | Expected behavior
-- | --
Cmd/Ctrl + Enter | Fill the selected range of cells with the value of the active cell 

The limitations and/or additional new behavior for the above keyboard shortcuts:
 * The filling works only when the last selection range (it matters for non-contiguous or multiple range selections) includes at least 2 cells. 
The example:
Kapture 2022-04-12 at 14 00 17
Here, the last selection is too small to trigger the filling (although the previous layers meet the requirements):
Kapture 2022-04-12 at 14 02 05

### How has this been tested?

I tested the changes locally and covered the new behavior with new E2E tests.

### Types of changes

- [x] New feature or improvement (non-breaking change which adds functionality)

### Related issue(s):
1. fixes #9366

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Review by kirszenbaum — APPROVED** [review]
I'll document these changes in #9271

File: handsontable/src/core.js [hunk 1]
```diff
@@ -4496,6 +4496,28 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
...
+  }, {
+    keys: [['Control/Meta', 'Enter']],
+    callback: () => {
+      const selectedRange = instance.getSelectedRange();
+      const {
+        row: highlightRow,
+        col: highlightColumn,
+      } = selectedRange[selectedRange.length - 1].highlight;
+      const valueToPopulate = instance.getDataAtCell(highlightRow, highlightColumn);
+      const cellValues = new Map();
```
Comment (wszymanski) [inline handsontable/src/core.js:4508]: Is the `Map` interface needed? Maybe simple array structure will be sufficient?

**Review by wszymanski — APPROVED** [review]
Good job, just one question.

File: handsontable/src/core.js (hunk 1 above)
Comment (budnix) [reply handsontable/src/core.js:4508]: There is a reason behind that choice. The Map unifies the keys put into the collection, so there is no coords duplication in the „changes” hooks.

Comment (wszymanski) [reply handsontable/src/core.js:4508]: Sound reasonable. Maybe extra comment/changing name (or both) to `unifiedCellValues` will protect it against refactor :)? The decision is up to you - PR looks good :white_check_mark:

**Review by aninde — CHANGES_REQUESTED** [review]
E2e tests include:
checking the getCellsCount method,
no population after selecting multi cell selection,
no population when selections overlap
population for range selection for the row and for the column,
the population for the selection obtained by clicking in the corner,
population for selection after clicking in row and column header

My tests involved checking `cmd / ctrl + Enter` on [macOS](https://app.qase.io/run/HOT/dashboard/79) and [Windows](https://app.qase.io/run/HOT/dashboard/80) for the editor and for various selections, including fixed, merged cell, cell with comment and in the read-only mode, hidden column and rows.

I was able to find issue here. Read-only cells can be populated after `Cmd/Ctrl + Enter` on selection, which I take as a mistake.

Kapture 2022-04-20 at 14 32 28

In an analogous situation, when in Google spreadsheet cells are given `Protected range`, when trying to population ears using the shortcut Cmd + Enter, a pop up with warning appears, as when trying to edit,  when such a setting is configured or nothing happend. 
Screenshot 2022-04-20 at 14 35 09

https://support.google.com/docs/answer/1218656?hl=en&co=GENIE.Platform%3DDesktop

**Comment by budnix** [comment]
> I was able to find issue here. Read-only cells can be populated after `Cmd/Ctrl + Enter` on selection, which I take as a mistake.

Thanks for catching that. I've prepared a fix within https://github.com/handsontable/handsontable/pull/9390/commits/5e9b03ad3b5d5f82160d13c62f8efe64c399eaf9. Can you re-check the issue again?

**Comment by aninde** [comment]
Also, I noticed that Option + Enter used on the editor still inserts new line on macOS.
Kapture 2022-04-20 at 14 03 39

**Comment by budnix** [comment]
> Also, I noticed that Option + Enter used on the editor still inserts new line on macOS.

Referring to the spec it should :)

**Review by aninde — APPROVED** [review]
Thank for instant fix! It works great :)

**Pull request merged**